sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +765,221 @@ def limiting_distribution(self):
         eqs.append(sum(wi) - 1)
         soln = list(linsolve(eqs, wi))[0]
         return ImmutableMatrix([[sol for sol in soln]])
+
+
+class BernoulliProcess(DiscreteTimeStochasticProcess):
+    """
+    The Bernoulli process consists of repeated
+    independent Bernoulli trials with the same parameter `p`.
+    It's assumed that the probability `p` applies to every
+    trial and that the outcomes of each trial
+    are independent of all the rest. Therefore Bernoulli Processs
+    is Discrete State and Discrete Time Stochastic Process.
+
+    Parameters
+    ==========
+
+    sym: Symbol/string_types
+    success: Event of success
+    failure: Event of failure
+    p: Proability of getting success
+
+    Examples
+    ========
+
+    >>> from sympy.stats import BernoulliProcess, P, E
+    >>> from sympy import Eq, Gt, Lt
+    >>> B = BernoulliProcess("B",p=0.7, success=1, failure=0)
+
+    >>> B.state_space
+    FiniteSet(0, 1)
+
+    >>> (B.p).round(2)
+    0.70
+
+    >>> B.success
+    1
+
+    >>> B.failure
+    0
+
+    >>> X = B[1] + B[2] + B[3]
+    >>> P(Eq(X, 0)).round(2)
+    0.03
+
+    >>> P(Eq(X, 2)).round(2)
+    0.44
+
+    >>> P(Eq(X, 4)).round(2)
+    0
+
+    >>> P(Gt(X, 1)).round(2)
+    0.78
+
+    >>> P(Eq(B[1], 0) & Eq(B[2], 1) & Eq(B[3], 0) & Eq(B[4], 1)).round(2)
+    0.04
+
+    >>> B.joint_distribution(B[1], B[2])
+    JointDistributionHandmade(Lambda((B[1], B[2]), Piecewise((0.7, Eq(B[1], 1)),
+    (0.3, Eq(B[1], 0)), (0, True))*Piecewise((0.7, Eq(B[2], 1)), (0.3, Eq(B[2], 0)),
+    (0, True))))
+
+    >>> E(2*B[1] + B[2]).round(2)
+    2.10
+
+    >>> P(B[1] < 1).round(2)
+    0.30
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Bernoulli_process
+    .. [2] https://mathcs.clarku.edu/~djoyce/ma217/bernoulli.pdf
+
+    """
+
+    index_set = S.Naturals0
+
+    @property
+    def p(self):
+        return self.args[3]
+
+    @property
+    def distribution(self):
+        return self.args[2]
+
+    @property
+    def success(self):
+        return self.args[4]
+
+    @property
+    def failure(self):
+        return self.args[5]
+
+    def __new__(cls, sym, p, success=1, failure=0):
+        if p > 1 or p < 0:
+            raise ValueError("Probability must be in between 0 and 1")
```

Comment:
What happens here if `p` is a symbol (e.g. `Symbol('p')`). I'd expect this to raise in that case:
```python
>>> from sympy.abc import p
>>> p > 1 or p < 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/core/relational.py", line 385, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```

Reply:
Probably, `_value_check` can be used here.